proto/auth.proto

```diff
@@ -69,8 +69,37 @@ message _GenerateApiTokenRequest {
         Expires expires = 2;
     }
 
-    // session token recieved from `momento login` command
-    string session_token = 3;
+    // Aliases for categories of functionality.
+    enum Role {
+      PermitNone = 0;
+      // Restricts access to apis that read and write data: No higher level resource description or modification.
+      ReadWrite = 1;
+    }
+
+    string auth_token = 3;
+
+    // Contains the restrictions that the token will have for each of our services. If it doesn't exist
+    // on the request, then the token is assumed to have "SuperUser" permissions
+    message Restrictions {
+      message CacheRestrictions {
+        Role role = 1;
+        // eventually could scope this down to a specific cache name or key
+        // string cache = "an example cache";
+        // string caches = ["cache1", "cache2"]
+      }
+
+      message TopicsRestrictions {
+        Role role = 1;
+        // eventually could scope this down to a specific topic name
+        // string topic = "an example topic";
+        // string topics = ["topic1", "topic2"]
+      }
+
+      CacheRestrictions cache = 1;
+      TopicsRestrictions topic = 2;
+    }
+
+    repeated Restrictions restrictions = 4;
```

Comment:
```suggestion
      oneof kind {
          CacheRestrictions cache = 1;
          TopicsRestrictions topic = 2;
      }
    }

    repeated Restrictions restrictions = 4;
```

since this is repeated now, this will be more flexible and should be significantly easier to parse (existence test won't be weird) without making it any harder to produce it.

Reply:
ah yea thats true, good call